server.js

```diff
@@ -1,35 +1,197 @@
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const nomineeSchema = new mongoose.Schema({
+  year_film: Number,
+  year_award: Number,
+  ceremony: Number,
+  category: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Category'
+  },
+  nominee: String,
+  film: String,
+  win: Boolean
+})
+
+const categorySchema = new mongoose.Schema({
+  description: String
+})
+const Category = mongoose.model('Category', categorySchema)
+
+const Nominee = mongoose.model('Nominee', nomineeSchema)
+
+if (process.env.RESET_D) {
+  const seedDB = async () => {
+    await Nominee.deleteMany()
+    await Category.deleteMany()
+
+    const categoriesArray = []
+
+    categoriesJason.forEach(async (item) => {
+      const category = new Category(item)
+      categoriesArray.push(category)
+      await category.save()
+    })
+
+    await goldenGlobesData.forEach(async (nominee) => {
+      const newNominee = new Nominee({
+        ...nominee,
+        // eslint-disable-next-line max-len
+        category: categoriesArray.find((oneCategory) => oneCategory.description === nominee.category)
+      })
+      await newNominee.save()
+    })
+  }
+  seedDB()
+}
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app));
+})
+
+app.get('/categories', async (req, res) => {
+  const categories = await Category.find()
+  res.json(categories)
+})
+
+app.get('/categories/:categoryId/nominees', async (req, res) => {
+  const { categoryId } = req.params
+  const { awardYear, nominee, film, win } = req.query
+  
+  try {
+    const query = {}
+    query.category = categoryId
+    if (awardYear) {
+      query.year_award = awardYear
+    }
+    if (nominee) {
+      query.nominee = { $regex: new RegExp(nominee, "i") }
+    }
+    if (film) {
+      query.film = { $regex: new RegExp(film, "i") }
+    }
+    if (win) {
+      query.win = win
+    }
+    
+    const nomineesCategory = await Nominee.find(query).populate('category')
+    if (nomineesCategory) {
+      res.json(nomineesCategory);
+    } else {
+      res.status(404).json({ error: "Not found" })
+    }
+  } catch (error) {
+    res.status(400).json({ error })
+  }
+});
+
+app.get('/nominees', async (req, res) => {
+  const { awardYear, nominee, page, perPage } = req.query
+  try {
+    const query = {}
+    if (awardYear) {
+      query.year_award = awardYear
+    }
+    if (nominee) {
+      query.nominee = { $regex: new RegExp(nominee, "i") }
+    }
+    const nominees = await Nominee.find(query).populate('category').skip(((Number(page) - 1) * Number(perPage))).limit(Number(perPage))
+    res.json(nominees);
```

Comment:
Maybe add a length so the front-end can see how many elements it is in the data. Just an idea 👯 

`res.json({ length: nominees.length, data: nominees });`